shpc/utils/fileio.py

```diff
@@ -14,6 +14,20 @@
 from shpc.logger import logger
 
 
+def can_be_deleted(path):
+    """
+    A path can be deleted if it contains no entries, *or*
+    if the only entry is '.version'
+    """
+    if os.path.exists(path):
+        entries = os.listdir(path)
+        if not entries:
+            return True
+        if entries == [".version"]:
+            return True
+    return False
+
+
```

Comment:
If we want this in utils, it probably should be general (and not customized to one specific use case). E.g.,:
```suggestion
def can_be_deleted(path, ignore_files=None):
    """
    A path can be deleted if it contains no entries, *or*
    if the only files are in ignore_files
    """
    ignore_files = ignore_files or []
    if os.path.exists(path):
        entries = os.listdir(path)
        if not entries:
            return True
        if entries == ignore_files:
            return True
    return False
```
And probably we'd either want to sort both lists, or do an iterator `[x for x in entries if x not in ignore_files]` and then return False if we find any remaining files in that set.